Packs/RetarusSecureEmailGateway/Integrations/RetarusSecureEmailGateway/RetarusSecureEmailGateway.py

```diff
@@ -0,0 +1,248 @@
+    
+
+@contextmanager
+def websocket_connection(url: str, token_id: str, fetch_interval: int, channel: str, verify_ssl: bool):
+    """
+    Create a connection to the api.
+
+    Args:
+        url (str): host URL for the websocket connection.
+        channel (str): Retarus channel to connect through.
+        token_id (str): Retarus token id.
+        fetch_interval (int): Time between fetch iterations.
+        verify_ssl (bool): Whether to verify ssl when connecting.
+
+    Yields:
+        EventConnection: eventConnection to receive events from.
+    """
+    extra_headers = {"Authorization": f"Bearer {token_id}"}
+    
+    ssl_context = ssl.create_default_context()
+    if not verify_ssl:
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+    
+    try:
+        with connect("wss://"+url+f"/email/siem/v1/websocket?channel={channel}",
+                     additional_headers=extra_headers,
+                     ssl=ssl_context) as ws:
+            connection = EventConnection(
+                connection=ws,
+                fetch_interval=fetch_interval
+            )
+            yield connection
+
+    except Exception as e:
+        raise DemistoException(f"{str(e)}\nAuthentication failed. Please check the URL, channel name and access token.")
+
+
+def is_interval_passed(fetch_start_time: datetime, fetch_interval: int) -> bool:
+    """This function checks if the given interval has passed since the given start time
+
+    Args:
+        fetch_start_time (datetime): The start time of the interval
+        fetch_interval (int): The interval in seconds
+
+    Returns:
+        bool: True if the interval has passed, False otherwise
+    """
+    return fetch_start_time + timedelta(seconds=fetch_interval) < datetime.now().astimezone(timezone.utc)
+
+
+def perform_long_running_loop(connection: EventConnection, fetch_interval: int):
+    """
+    Long running loop iteration function. Fetches events from the connection and sends them to XSIAM.
+
+    Args:
+        connection (EventConnection): A connection object to fetch events from.
+        fetch_interval (int): Fetch time for this fetching events cycle.
+    """
+    integration_context = demisto.getIntegrationContext()
+    events = fetch_events(connection, fetch_interval)
+    events.extend(integration_context.get("events", []))
+    integration_context["events"] = events  # update events in context in case of a failure.
+    demisto.debug(f'{LOG_PREFIX} Adding {len(events)} Events to XSIAM')
+    
+    # Send the events to the XSIAM, with events from the context
+    # Need to add the option that if we have more then one failure of sending the events to xsiam then we stop fetching. consult with Meital and Dima # TODO
+    try:
+        send_events_to_xsiam(events, vendor=VENDOR, product=PRODUCT)
+        # clear the context after sending the events
+        demisto.setIntegrationContext({})
```

Comment:
Its a better practice to decide on a specific key and update it and not the whole context as it is preventing future use.